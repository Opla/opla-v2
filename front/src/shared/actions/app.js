/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import {
  APP_DELETEINTENTACTION,
  APP_SELECTINTENT,
  APP_SELECTIO,
  APP_UNSELECTIO,
  APP_SELECTENTITY,
  APP_SELECTFUNCTION,
  APP_SETINTENTACTION,
  APP_UPDATEINTENT,
  APP_UPDATEPUBLISHER,
  APP_SETNEWACTIONS,
  APP_DELETENEWACTIONS,
} from "./constants";

export function appSelectIntent(selectedBotId, selectedIntentIndex) {
  return { type: APP_SELECTINTENT, selectedBotId, selectedIntentIndex };
}

export function appSelectIO(selectedIntentIndex, selectedIOIndex, isOutput) {
  return { type: APP_SELECTIO, selectedIntentIndex, selectedIOIndex, isOutput };
}

export function appUnSelectIO() {
  return { type: APP_UNSELECTIO };
}

export function appUpdatePublisher(selectedBotId, publisher) {
  return { type: APP_UPDATEPUBLISHER, selectedBotId, publisher };
}

export function appUpdateIntent(selectedBotId, intent) {
  return { type: APP_UPDATEINTENT, selectedBotId, intent };
}

export function appSetIntentAction(
  actionContainer,
  actionType,
  actionValue,
  selectedAction,
) {
  return {
    type: APP_SETINTENTACTION,
    actionContainer,
    actionType,
    actionValue,
    selectedAction,
  };
}

export function appDeleteIntentAction(actionContainer, selectedAction) {
  return { type: APP_DELETEINTENTACTION, actionContainer, selectedAction };
}

export function appSetNewActions(actionContainer, actionValue) {
  return {
    type: APP_SETNEWACTIONS,
    actionContainer,
    actionValue,
  };
}

export function appDeleteNewActions() {
  return {
    type: APP_DELETENEWACTIONS,
  };
}

export function appSelectEntity(selectedBotId, selectedEntityIndex) {
  return { type: APP_SELECTENTITY, selectedBotId, selectedEntityIndex };
}

export function appSelectFunction(selectedBotId, selectedFunctionIndex) {
  return { type: APP_SELECTFUNCTION, selectedBotId, selectedFunctionIndex };
}
