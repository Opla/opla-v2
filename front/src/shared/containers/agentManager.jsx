/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Grid, Inner, Cell } from "zrmc";
import Loading from "zoapp-front/components/loading";
import SignInForm from "zoapp-front/containers/signInForm";

import ExplorerContainer from "./explorerContainer";
import IntentContainer from "./builder/intentContainer";
import EntityContainer from "./builder/entityContainer";
import FunctionContainer from "./builder/functionContainer";

class AgentManager extends Component {
  render() {
    let { isLoading } = this.props;
    if (!isLoading && !this.props.intents && this.props.isSignedIn) {
      isLoading = true;
    }
    if (!this.props.isSignedIn) {
      return <SignInForm />;
    } else if (this.props.intents == null) {
      return <Loading />;
    }
    let panel1 = null;
    let panel2 = null;

    panel1 = (
      <Cell
        style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
        className="zui-color--white zui-panel"
        span={3}
      >
        <ExplorerContainer
          onRename={this.props.onRenameIntent}
          onAdd={this.props.onAddIntent}
          onDelete={this.props.onDeleteIntent}
        />
      </Cell>
    );
    if (this.props.selectedType === "function") {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <FunctionContainer />
        </Cell>
      );
    } else if (this.props.selectedType === "entity") {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <EntityContainer />
        </Cell>
      );
    } else {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <IntentContainer handleRename={this.handleRenameIntent} />
        </Cell>
      );
    }
    return (
      <Grid
        gutter={{ desktop: "0px", tablet: "0px", phone: "0px" }}
        style={{ margin: "0px", padding: "0px" }}
      >
        <Inner style={{ gridGap: "0px" }}>
          {panel1}
          {panel2}
        </Inner>
      </Grid>
    );
  }
}

AgentManager.defaultProps = {
  bot: null,
  intents: null,
  selectedIntent: null,
  selectedBotId: null,
  store: null,
  activeTab: 0,
};

AgentManager.propTypes = {
  activeTab: PropTypes.number,
  isLoading: PropTypes.bool.isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  selectedBotId: PropTypes.string,
  selectedType: PropTypes.string.isRequired,
  bot: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
  }),
  intents: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string })),
  selectedIntentIndex: PropTypes.number.isRequired,
  selectedIntent: PropTypes.shape({ id: PropTypes.string }),
  store: PropTypes.shape({}),
  titleName: PropTypes.string.isRequired,

  onRenameIntent: PropTypes.func.isRequired,
  onAddIntent: PropTypes.func.isRequired,
  onDeleteIntent: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  let { selectedIntent } = state.app;
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  // TODO get selectedBot from selectBotId
  const bot = selectedBotId ? admin.bots[0] : null;
  const intents = state.app.intents ? state.app.intents : null;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading || false;
  const selectedIntentIndex = state.app ? state.app.selectedIntentIndex : 0;
  const selectedType = state.app ? state.app.selectedType : "intent";
  if (!selectedIntent) {
    selectedIntent = state.app.intents
      ? state.app.intents[selectedIntentIndex]
      : null;
  }
  const titleName = state.app.activeScreen.name
    ? state.app.activeScreen.name
    : "";
  return {
    selectedBotId,
    bot,
    intents,
    isLoading,
    isSignedIn,
    selectedIntent,
    selectedIntentIndex,
    selectedType,
    titleName,
  };
};

const mapDispatchToProps = () => ({});

// prettier-ignore
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AgentManager);
