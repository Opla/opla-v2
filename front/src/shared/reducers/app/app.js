/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  /* APP Section */
  APP_SELECTENTITY,
  APP_SELECTFUNCTION,
  APP_SELECTVARIABLE,
} from "../../actions/constants";

export const initialState = {
  selectedType: "intent",
  selectedEntityIndex: -1,
  selectedFunctionIndex: -1,
};

export const handlers = {
  /* APP Section */

  // SELECT ENTITY
  [APP_SELECTENTITY]: (state, { selectedBotId, selectedEntityIndex }) => {
    const selectedIntent = null;
    const selectedIntentIndex = -1;
    const selectedVariableIndex = -1;
    const selectedFunctionIndex = -1;
    const selectedType = "entity";
    return {
      ...state,
      selectedBotId,
      selectedIntentIndex,
      selectedIntent,
      selectedEntityIndex,
      selectedVariableIndex,
      selectedFunctionIndex,
      selectedType,
    };
  },
  // SELECT VARIABLE
  [APP_SELECTVARIABLE]: (state, { selectedBotId, selectedVariableIndex }) => {
    const selectedIntent = null;
    const selectedIntentIndex = -1;
    const selectedFunctionIndex = -1;
    const selectedEntityIndex = -1;
    const selectedType = "variable";
    return {
      ...state,
      selectedBotId,
      selectedIntentIndex,
      selectedIntent,
      selectedEntityIndex,
      selectedVariableIndex,
      selectedFunctionIndex,
      selectedType,
    };
  },
  [APP_SELECTFUNCTION]: (state, { selectedBotId, selectedFunctionIndex }) => {
    const selectedIntent = null;
    const selectedIntentIndex = -1;
    const selectedEntityIndex = -1;
    const selectedVariableIndex = -1;
    const selectedType = "function";
    return {
      ...state,
      selectedBotId,
      selectedIntentIndex,
      selectedIntent,
      selectedEntityIndex,
      selectedVariableIndex,
      selectedFunctionIndex,
      selectedType,
    };
  },
};
