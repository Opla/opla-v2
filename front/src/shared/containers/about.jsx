/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { appSetTitleName } from "zoapp-front/dist/actions/app";

const aboutStyle = {
  margin: "5% 40% 10% 10%",
};

class About extends Component {
  constructor(props) {
    super();
    props.appSetTitleName("About");
  }

  render() {
    const { isLoading } = this.props;

    return (
      <div style={aboutStyle}>
        {isLoading ? "" : null}
        <h1>Essential</h1>
        <h4>Your own virtual assistant</h4>
        <p>
          We are Opla, the mighty volcanic team! Since 2015 we started
          developing this deeptech chatbot solution. Nowadays we are truly
          commited to OpenSource and fully involved in creating the most
          complete and user oriented conversational interface tool.
        </p>
        <h1>Version</h1>
        <h4>V0.1 - Amazing Alou </h4>
        <p>
          (Le Puy Alou : a french volcano) July 2018<br />
          <a
            href="https://github.com/Opla/"
            target="_blank"
            rel="noreferrer noopener"
          >
            See more on Github
          </a>
        </p>
        <h1>Contributors</h1>
        <h4>Thanks to the team and contributors</h4>
        <p>
          Elodie, Julie, Maali, Marie, Nicolle, Adil, Dorian, Florian, Loic,
          Manu, Mik, Thomas, Will, Will<br />
          <a
            href="https://github.com/Opla/joinUs"
            target="_blank"
            rel="noreferrer noopener"
          >
            And we are hiring !
          </a>
        </p>
        <p>
          Follow us on Twitter
          <a
            href="https://twitter.com/opla_ai/"
            target="_blank"
            rel="noreferrer noopener"
          >
            @opla_ai
          </a>
          <br />
          Or contact us hello@opla.ai
        </p>
      </div>
    );
  }
}

About.propTypes = {
  isLoading: PropTypes.bool,
  appSetTitleName: PropTypes.func.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  appSetTitleName: (titleName) => {
    dispatch(appSetTitleName(titleName));
  },
});

// prettier-ignore
export default connect(
  null,
  mapDispatchToProps,
)(About);
