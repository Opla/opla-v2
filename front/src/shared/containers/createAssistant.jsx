/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, { Select, MenuItem, Button, TextField } from "zrmc";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import ProcessingDialog from "zoapp-front/dist/containers/processingDialog";
import { appSetTitleName, setMessage } from "zoapp-front/dist/actions/app";

import TemplatesList from "../components/templatesList";
import {
  apiCreateBot,
  apiGetTemplatesRequest,
  apiGetLanguagesRequest,
} from "../actions/api";

const boxStyle = {
  margin: "16px",
};

const headerStyle = {
  padding: "16px",
};

const h4 = {
  marginTop: "0px",
  marginBottom: "16px",
  fontSize: "16px",
  color: "rgba(0, 0, 0, 0.87)",
};
const secText = {
  color: "rgba(0, 0, 0, 0.54)",
};

export class CreateAssistantBase extends Component {
  constructor(props) {
    super();
    this.state = {
      name: "",
      language: null,
      loading: false,
      selectedTemplate: null,
      template: null,
    };
    props.appSetTitleName("Create an assistant", "Create");
  }

  componentDidMount() {
    this.props.apiGetTemplates();
    this.props.apiGetLanguages();
  }

  componentDidUpdate(prevProps) {
    if (
      prevProps.templates !== this.props.templates &&
      this.state.selectedTemplate === null
    ) {
      const selectedTemplate = this.props.templates.findIndex(
        (template) => template.name === "Empty",
      );

      this.onSelectTemplate(
        selectedTemplate,
        this.props.templates[selectedTemplate],
      );
    }
    if (this.state.loading && this.props.isLoading === false) {
      this.handleCloseCreateDialog();
    }
  }

  onImportTemplate = (data) => {
    try {
      const json = JSON.parse(data);
      this.onSelectTemplate(3, json);
    } catch (e) {
      this.props.setMessage("imported template is not a valid JSON document");
    }
  };

  onSelectTemplate = (selected, data) => {
    let template = data;
    if (!template) {
      template = this.props.templates[selected];
    }
    this.setState({ selectedTemplate: selected, template });
  };

  handleCloseCreateDialog = () => {
    this.setState({ loading: false });
    Zrmc.closeDialog();
    if (!this.props.error) {
      // TODO select Bot created
      this.props.history.push("/factory");
    }
  };

  handleCreate = (e) => {
    e.preventDefault();

    const { name, email, language, template, loading } = this.state;

    if (template === null) {
      this.props.setMessage("please select a template");
      return;
    }

    if (loading === false) {
      const botParams = {
        name,
        email,
        template,
        language,
      };

      this.setState({ loading: true });

      Zrmc.showDialog(
        <ProcessingDialog open onClosed={this.handleCloseDialog} />,
      );

      this.props.createBot(botParams);
    } else {
      // TODO display errors in dialogs
    }
  };

  handleLanguageChange = (language) => {
    this.setState({ language });
  };

  createChangeHandler = (field) => (e) => {
    this.setState({ [field]: e.target.value });
  };

  render() {
    const { name, email, selectedTemplate: selected } = this.state;

    let selectedLanguageIndex = 0;
    const languagesItems = this.props.languages.map((language, index) => {
      if (language.default === true) {
        selectedLanguageIndex = index;
      }
      return (
        <MenuItem
          key={language.id}
          selected={language.default}
          value={language.id}
        >
          {language.name}
        </MenuItem>
      );
    });

    // TODO json only for instance
    const acceptImport = "application/json";
    return (
      <div className="zui-layout__content zui-color--grey-100">
        <section className="zui-color--white zui-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Welcome</h4>
            <div style={secText}>
              Create your open conversational robot at the speed of light. Just
              follow the instructions, no coding required !
            </div>
          </div>
        </section>
        <section className="zui-color--white zui-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Templates</h4>
            <div style={secText}>
              Choose a prebuild asssistant, import one or select an empty model.
            </div>
          </div>
          {this.props.templates &&
            this.props.templates.length > 0 && (
              <TemplatesList
                items={this.props.templates}
                selectedItem={selected}
                onSelect={this.onSelectTemplate}
                onImport={this.onImportTemplate}
                acceptImport={acceptImport}
              />
            )}
        </section>
        <form
          id="create-assistant-form"
          autoComplete="new-password"
          onSubmit={this.handleCreate}
        >
          <section
            className="zui-color--white zui-shadow--2dp"
            style={boxStyle}
          >
            <div style={headerStyle}>
              <h4 style={h4}>Informations needed</h4>
              <div style={secText}>
                Please provide the following information. Don&apos;t worry. You
                can always change them later.
              </div>
              <div>
                <TextField
                  id="create-assistant-name"
                  onChange={this.createChangeHandler("name")}
                  defaultValue={name}
                  label="Assistant name"
                  style={{ width: "400px" }}
                  required
                />
              </div>
              <div>
                <TextField
                  id="create-assistant-email"
                  type="email"
                  onChange={this.createChangeHandler("email")}
                  defaultValue={email}
                  label="Contact email"
                  style={{ width: "400px" }}
                  required
                />
              </div>
              <div>
                <Select
                  id="create-assistant-language"
                  label="Choose language"
                  onSelected={this.handleLanguageChange}
                  style={{ width: "400px" }}
                  selectedIndex={selectedLanguageIndex}
                  required
                >
                  {languagesItems}
                </Select>
              </div>
            </div>
          </section>
          <section style={boxStyle}>
            <div style={headerStyle}>
              <Button type="submit" raised>
                Create
              </Button>
            </div>
          </section>
        </form>
      </div>
    );
  }
}

CreateAssistantBase.defaultProps = {
  error: null,
};

CreateAssistantBase.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({})]),
  createBot: PropTypes.func.isRequired,
  appSetTitleName: PropTypes.func.isRequired,
  apiGetTemplates: PropTypes.func.isRequired,
  apiGetLanguages: PropTypes.func.isRequired,
  setMessage: PropTypes.func.isRequired,
  history: PropTypes.shape({ length: PropTypes.number, push: PropTypes.func })
    .isRequired,
  templates: PropTypes.array,
  languages: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  const { admin, error, templates, languages } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.app.loading || false;
  return {
    admin,
    isLoading,
    isSignedIn,
    error,
    templates,
    languages,
  };
};

const mapDispatchToProps = (dispatch) => ({
  createBot: (botParams) => {
    dispatch(apiCreateBot(botParams));
  },
  appSetTitleName: (titleName) => {
    dispatch(appSetTitleName(titleName));
  },
  setMessage: (message) => {
    dispatch(setMessage(message));
  },
  apiGetTemplates: () => {
    dispatch(apiGetTemplatesRequest());
  },
  apiGetLanguages: () => {
    dispatch(apiGetLanguagesRequest());
  },
});

// prettier-ignore
export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(CreateAssistantBase),
);
