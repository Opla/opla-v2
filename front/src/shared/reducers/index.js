/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { combineReducers } from "redux";
import auth from "zoapp-front/dist/reducers/auth";
import initialize from "zoapp-front/dist/reducers/initialize";
import user from "zoapp-front/dist/reducers/user";

import app from "./app";
import message from "./message";
import metrics from "./metrics";

export default combineReducers({
  initialize,
  auth,
  user,
  metrics,
  app,
  message,
});
