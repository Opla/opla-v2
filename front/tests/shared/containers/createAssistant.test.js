/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import { createFakeEvent } from "tests/helpers";
import { CreateAssistantBase } from "shared/containers/createAssistant";
import { defaultTemplates, defaultLanguages } from "shared/reducers/app";
import Zrmc from "zrmc";

beforeAll(() => {
  Zrmc.showDialog = jest.fn();
});

describe("containers/CreateAssistant", () => {
  it("renders correctly", () => {
    const appSetTitleNameSpy = jest.fn();
    const setMessage = jest.fn();
    const createBotSpy = jest.fn();
    const apiGetTemplatesSpy = jest.fn();
    const apiGetLanguagesSpy = jest.fn();
    const historySpy = { length: 0, push: jest.fn() };

    const component = renderer.create(
      <CreateAssistantBase
        isLoading={false}
        createBot={createBotSpy}
        appSetTitleName={appSetTitleNameSpy}
        apiGetTemplates={apiGetTemplatesSpy}
        apiGetLanguages={apiGetLanguagesSpy}
        templates={defaultTemplates}
        languages={defaultLanguages}
        history={historySpy}
        setMessage={setMessage}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(appSetTitleNameSpy).toHaveBeenCalled();
    expect(apiGetTemplatesSpy).toHaveBeenCalled();
    expect(apiGetLanguagesSpy).toHaveBeenCalled();
    expect(createBotSpy).not.toHaveBeenCalled();
    expect(historySpy.push).not.toHaveBeenCalled();
    expect(setMessage).not.toHaveBeenCalled();
  });

  it("should update template selected when props change", () => {
    const onSelectTemplateSpy = jest.fn();
    const newPropsTemplates = [
      { id: "abcdefgh", name: "Foo" },
      { id: "abcdefgh-251a-4e11-a907-b1f3bcc20283", name: "Empty" },
    ];
    const wrapper = shallow(
      <CreateAssistantBase
        isLoading={false}
        createBot={jest.fn}
        appSetTitleName={jest.fn()}
        apiGetTemplates={jest.fn()}
        apiGetLanguages={jest.fn()}
        templates={defaultTemplates}
        languages={defaultLanguages}
        setMessage={jest.fn()}
        history={{ length: 0, push: jest.fn() }}
      />,
    );
    wrapper.instance().onSelectTemplate = onSelectTemplateSpy;
    wrapper.setProps({ templates: newPropsTemplates });
    expect(onSelectTemplateSpy).toHaveBeenCalledWith(1, newPropsTemplates[1]);
  });

  it("sets a message if uploaded template is not a valid json", () => {
    const setMessage = jest.fn();

    const wrapper = shallow(
      <CreateAssistantBase
        isLoading={false}
        createBot={jest.fn}
        appSetTitleName={jest.fn()}
        apiGetTemplates={jest.fn()}
        apiGetLanguages={jest.fn()}
        templates={defaultTemplates}
        languages={defaultLanguages}
        setMessage={setMessage}
        history={{ length: 0, push: jest.fn() }}
      />,
    );

    wrapper.instance().onImportTemplate("this is not a json obviously");

    expect(setMessage).toHaveBeenCalledWith(
      "imported template is not a valid JSON document",
    );
  });

  it("selects empty template once complete template list loaded", () => {
    const wrapper = shallow(
      <CreateAssistantBase
        isLoading={false}
        createBot={jest.fn}
        appSetTitleName={jest.fn()}
        apiGetTemplates={jest.fn()}
        apiGetLanguages={jest.fn()}
        templates={defaultTemplates}
        languages={defaultLanguages}
        setMessage={jest.fn()}
        history={{ length: 0, push: jest.fn() }}
      />,
    );

    const templates = defaultTemplates.slice();
    templates.push({
      id: "foo",
      name: "loaded template",
    });

    wrapper.setProps({ templates });

    expect(wrapper.state("template")).not.toBeNull();
    expect(wrapper.state("selectedTemplate")).not.toBeNull();
  });

  it("creates a bot with fullfilled form", () => {
    const createBotSpy = jest.fn();

    const wrapper = shallow(
      <CreateAssistantBase
        isLoading={false}
        createBot={createBotSpy}
        appSetTitleName={jest.fn()}
        setMessage={jest.fn()}
        apiGetTemplates={jest.fn()}
        apiGetLanguages={jest.fn()}
        templates={defaultTemplates}
        languages={defaultLanguages}
        history={{ length: 0, push: jest.fn() }}
      />,
    );

    wrapper
      .find("#create-assistant-name")
      .simulate("change", { target: { value: "assistant-name" } });
    expect(wrapper.state("name")).toEqual("assistant-name");

    wrapper
      .find("#create-assistant-username")
      .simulate("change", { target: { value: "assistant-username" } });
    expect(wrapper.state("username")).toEqual("assistant-username");

    wrapper
      .find("#create-assistant-password")
      .simulate("change", { target: { value: "assistant-password" } });
    expect(wrapper.state("password")).toEqual("assistant-password");

    wrapper
      .find("#create-assistant-email")
      .simulate("change", { target: { value: "fake-email@opla.ai" } });
    expect(wrapper.state("email")).toEqual("fake-email@opla.ai");

    wrapper.find("#create-assistant-language").simulate("selected", "fr");
    expect(wrapper.state("language")).toEqual("fr");

    wrapper.instance().onSelectTemplate(0);

    wrapper
      .find("#create-assistant-form")
      .simulate("submit", createFakeEvent());

    expect(wrapper.state("loading")).toEqual(true);
    expect(createBotSpy).toHaveBeenCalled();
  });
});
