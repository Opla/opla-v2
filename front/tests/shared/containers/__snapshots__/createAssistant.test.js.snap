// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CreateAssistant renders correctly 1`] = `
<div
  className="zui-layout__content zui-color--grey-100"
>
  <section
    className="zui-color--white zui-shadow--2dp"
    style={
      Object {
        "margin": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <h4
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": "16px",
            "marginBottom": "16px",
            "marginTop": "0px",
          }
        }
      >
        Welcome
      </h4>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        Create your open conversational robot at the speed of light. Just follow the instructions, no coding required !
      </div>
    </div>
  </section>
  <section
    className="zui-color--white zui-shadow--2dp"
    style={
      Object {
        "margin": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <h4
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": "16px",
            "marginBottom": "16px",
            "marginTop": "0px",
          }
        }
      >
        Templates
      </h4>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        Choose a prebuild asssistant, import one or select an empty model.
      </div>
    </div>
    <div
      className="mdc-layout-grid"
    >
      <div
        className="mdc-layout-grid__inner"
      >
        <div
          className="opla_assistant_card-box mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet mdc-layout-grid__cell--span-6-phone mdc-layout-grid__cell--order-1"
        >
          <div
            onClick={[Function]}
            role="presentation"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="opla_assistant_card-container selectableListItem"
            >
              <div
                className="opla_assistant_card-imgContainer"
              >
                <div
                  className="opla_assistant_card-background"
                  style={
                    Object {
                      "backgroundColor": "#4cdbc4",
                    }
                  }
                >
                  <div
                    className="opla_assistant_card-imgMask"
                  >
                    <img
                      alt="Empty"
                      className="opla_assistant_card-img"
                      src="./images/robots/robot-0.svg"
                    />
                  </div>
                </div>
              </div>
              <div
                className="opla_assistant_card-text"
              >
                <b>
                  Empty
                </b>
              </div>
            </div>
          </div>
        </div>
        <div
          className="opla_assistant_card-box mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet mdc-layout-grid__cell--span-6-phone mdc-layout-grid__cell--order-2"
        >
          <div
            onClick={[Function]}
            role="presentation"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="opla_assistant_card-container selectableListItem"
            >
              <div
                className="opla_assistant_card-imgContainer"
              >
                <form>
                  <div
                    className="fileInput"
                  >
                    <i
                      className="material-icons"
                    >
                      cloud_upload
                    </i>
                    <input
                      accept="application/json"
                      onChange={[Function]}
                      type="file"
                    />
                  </div>
                </form>
              </div>
              <div
                className="opla_assistant_card-text"
              >
                <b>
                  Import
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <form
    autoComplete="new-password"
    id="create-assistant-form"
    onSubmit={[Function]}
  >
    <section
      className="zui-color--white zui-shadow--2dp"
      style={
        Object {
          "margin": "16px",
        }
      }
    >
      <div
        style={
          Object {
            "padding": "16px",
          }
        }
      >
        <h4
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": "16px",
              "marginBottom": "16px",
              "marginTop": "0px",
            }
          }
        >
          Informations needed
        </h4>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
            }
          }
        >
          Please provide the following information. Don't worry. You can always change them later.
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              className="mdc-text-field__input"
              defaultValue=""
              id="create-assistant-name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-name"
              style={Object {}}
            >
              Assistant name
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              className="mdc-text-field__input"
              defaultValue={undefined}
              id="create-assistant-email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-email"
              style={Object {}}
            >
              Contact email
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            aria-hidden={true}
            className="mdc-select"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="listbox"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <div
              className="mdc-select__native-control"
              tabIndex="0"
            >
              <div
                className="mdc-floating-label mdc-floating-label--float-above"
              >
                Choose language
              </div>
              <div
                className="mdc-select__selected-text"
              >
                English
              </div>
              <div
                className="mdc-line-ripple"
              />
            </div>
            <div
              className="mdc-select__menu mdc-menu"
              tabIndex="-1"
            >
              <ul
                aria-hidden="true"
                className="mdc-menu__items mdc-list"
                role="menu"
              >
                <li
                  aria-selected="true"
                  className="mdc-list-item"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  role="option"
                  tabIndex="0"
                >
                  English
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      style={
        Object {
          "margin": "16px",
        }
      }
    >
      <div
        style={
          Object {
            "padding": "16px",
          }
        }
      >
        <button
          className="mdc-button mdc-button--raised"
          disabled={false}
          id={null}
          onClick={null}
          type="submit"
        >
          
          Create
        </button>
      </div>
    </section>
  </form>
</div>
`;
